{{ cookiecutter.package_name }}/tox.ini

```diff
@@ -0,0 +1,82 @@
+[tox]
+envlist =
+    py{36,37,38}-test{,-alldeps,-oldestdeps,-devdeps,-numpy116,-numpy117,-numpy118}
+    build_docs
+    codestyle
+requires =
+    setuptools >= 30.3.0
+    pip >= 19.3.1
+isolated_build = true
+
+[testenv]
+
+# Pass through the following environemnt variables which may be needed for the CI
+passenv = HOME WINDIR LC_ALL LC_CTYPE CC CI TRAVIS
+
+# Run the tests in a temporary directory to make sure that we don't import
+# astropy from the source tree
+changedir = .tmp/{envname}
+
+# tox environments are constructued with so-called 'factors' (or terms)
+# separated by hyphens, e.g. test-devdeps-cov. Lines below starting with factor:
+# will only take effect if that factor is included in the environment name. To
+# see a list of example environments that can be run, along with a description,
+# run:
+#
+#     tox -l -v
+#
+description =
+    run tests
+    alldeps: with all optional dependencies
+    devdeps: with the latest developer version of key dependencies
+    oldestdeps: with the oldest supported version of key dependencies
+    cov: and test coverage
+    numpy116: with numpy 1.16.*
+    numpy117: with numpy 1.17.*
+    numpy118: with numpy 1.18.*
+
+deps =
+
+    numpy116: numpy==1.16.*
+    numpy117: numpy==1.17.*
+    numpy118: numpy==1.18.*
+
+    astropy30: astropy==3.0.*
```

Comment:
not exactly clear where you use these, don't you need to list it above in `envlist` as a pattern?

Reply:
You don't need to list all possible options in ``envlist`` - you can basically include ``astropy30`` in the env name anywhere if you want to pin it. So you could do e.g. ``py38-test-astropy30`` or ``py38-numpy117-astropy30-test`` (the order is not important). I can add them to the envlist though for clarity.

Reply:
ahh, ok. This just shows that I should have looked into tox a bit more.